packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2779,31 +3138,34 @@ export class TransactionController extends BaseControllerV1<
       return undefined;
     }
 
-    await this.updateTransactionMetaRSV(transactionMeta, signedTx);
-
-    transactionMeta.status = TransactionStatus.signed;
+    const transactionMetaWithRsv = {
+      ...(await this.updateTransactionMetaRSV(transactionMeta, signedTx)),
+      status: TransactionStatus.signed as const,
+    };
 
     this.updateTransaction(
-      transactionMeta,
+      transactionMetaWithRsv,
       'TransactionController#approveTransaction - Transaction signed',
     );
 
-    this.onTransactionStatusChange(transactionMeta);
+    this.onTransactionStatusChange(transactionMetaWithRsv);
 
     const rawTx = bufferToHex(signedTx.serialize());
 
-    transactionMeta.rawTx = rawTx;
+    const transactionMetaWithRawTx = { ...transactionMetaWithRsv, rawTx };
```

Comment:
since we shallow cloned the root object already, would it be okay to modify `transactionMetaWithRsv` directly?

Reply:
I'm not sure if the `updateTransaction` in between would freeze it again, but even so if the pattern is now to duplicate rather than mutate, i'd vote to avoid any mutation to maintain clarity.